airflow/models/dagrun.py

```diff
@@ -839,25 +839,59 @@ def task_filter(task: "Operator") -> bool:
 
         if hook_is_noop:
 
-            def create_ti_mapping(task: "Operator") -> dict:
-                created_counts[task.task_type] += 1
-                return TI.insert_mapping(self.run_id, task, map_index=-1)
+            def create_ti_mapping(task: "Operator", session: Session = session) -> List[dict]:
+
+                if isinstance(task, MappedOperator):
+                    indexes_to_map = task.get_indexes_to_map(self.run_id, session=session)
```

Comment:
We can only ever be able to expand tasks with literal-only kwargs at this point, so we probably don’t need to call `_get_map_lengths` (which is expensive and accesses db), but instead rely solely on the task kwargs to get this length.